native/core/src/parquet/mod.rs

```diff
@@ -650,21 +651,29 @@ pub unsafe extern "system" fn Java_org_apache_comet_parquet_Native_initRecordBat
     required_schema: jbyteArray,
     data_schema: jbyteArray,
     session_timezone: jstring,
+    batch_size: jint,
+    worker_threads: jint,
+    blocking_threads: jint,
 ) -> jlong {
     try_unwrap_or_throw(&e, |mut env| unsafe {
-        let task_ctx = TaskContext::default();
+        let session_config = SessionConfig::new().with_batch_size(batch_size as usize);
+        let planer =
```

Comment:
Do we need to create a new planner here? We really only need a session_config (even though it is really not valid for the entire session).

Reply:
This planner is used for native filter conversion

Reply:
Oh I see it now.